user-service/controller/user-controller.js

```diff
@@ -44,4 +48,62 @@ export async function loginUser(req, res) {
     } catch (err) {
         return res.status(500).json({ message: `Database failure when logging in as ${username}!` })
     }
-}
\ No newline at end of file
+}
+
+export async function forgetPassword(req, res) {
+    try {
+        const { username } = req.body;
+        // Check the existance of input
+        if (!username) {
+            return res.status(400).json({ message: 'Username is missing!' });
+        }
+
+        const resp = await _forgetPassword(username);
+        if(resp.err) {
+            return res.status(400).json({ message: resp.err });
+        } else {
+            const msg = resp.message;
+            console.log(msg);
+            return res.status(200).json({ msg });
+        }
+    } catch (err) {
+        return res.status(500).json({ message: `Database failure when sending reset link!` })
+    }
+}
+
+export async function verifyResetPassword(req, res) {
+    try {
+        const { username, token } = req.params;
+        const resp = await _verifyResetPassword(username, token);
+        
+        if (resp.err || !resp) {
+            return res.status(400).json({ message: resp.err });

```

Comment:
if resp is false. then there would be an error when returning resp.err

Reply:
I misunderstood the flow. Thank you, I will make the change.